proto/auth.proto

```diff
@@ -69,8 +69,34 @@ message _GenerateApiTokenRequest {
         Expires expires = 2;
     }
 
-    // session token recieved from `momento login` command
-    string session_token = 3;
+    enum Role {
+      ReadWrite = 0;
+    }
+
+    string auth_token = 3;
+
+    // Contains the permissions that the token will have for each of our services. If it doesn't exist
+    // on the request, then the token is assumed to have "SuperUser" permissions
+    message Permissions {
+      message CachePermissions {
```

Comment:
maybe this should be `CacheRestrictions`?

Reply:
imo that would be the thing to do.

Reply:
if we want to go with the language `restrictions` ok, but personally I find it breaks my brain a little? Using `permissions` here is more what I'm used to.